src/components/base/LegacyModal/index.js

```diff
@@ -0,0 +1,249 @@
+  }),
+})`
+  position: fixed;
+  z-index: 30;
+`
+
+const Backdrop = styled(Box).attrs({
+  bg: p => rgba(p.theme.colors.black, 0.4),
+  sticky: true,
+  style: p => ({
+    opacity: p.op,
+  }),
+})`
+  position: fixed;
+`
+
+const NonClickableHeadArea = styled.div`
+  position: fixed;
+  height: 48px;
+  width: 100%;
+  top: 0;
+  left: 0;
+  z-index: 1;
+`
+
+const Wrapper = styled(Box).attrs({
+  bg: 'transparent',
+  flow: 4,
+  style: p => ({
+    opacity: p.op,
+    transform: `scale3d(${p.scale}, ${p.scale}, ${p.scale})`,
+  }),
+})`
+  outline: none;
+  width: ${p => (p.width ? p.width : '500px')};
+  z-index: 2;
+`
+
+class Pure extends Component<any> {
+  shouldComponentUpdate(nextProps) {
+    if (nextProps.isAnimated) {
+      return false
+    }
+
+    return true
+  }
+
+  render() {
+    const { data, onClose, render } = this.props
+
+    return render({ data, onClose })
+  }
+}
+
+function stopPropagation(e) {
+  e.stopPropagation()
+}
+
+const wrap = EXPERIMENTAL_CENTER_MODAL
+  ? children => (
+      <Box alignItems="center" justifyContent="center" grow>
+        {children}
+      </Box>
+    )
+  : children => (
+      <GrowScroll alignItems="center" full pt={8}>
+        {children}
+      </GrowScroll>
+    )
+
+export class Modal extends Component<Props> {
+  static defaultProps = {
+    isOpened: false,
+    onHide: noop,
+    preventBackdropClick: false,
+  }
+
+  shouldComponentUpdate(nextProps: Props) {
+    if (this.props.isOpened || nextProps.isOpened) {
+      return true
+    }
+
+    return false
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const didOpened = this.props.isOpened && !prevProps.isOpened
+    const didClose = !this.props.isOpened && prevProps.isOpened
+    const shouldFocus = didOpened || this.props.refocusWhenChange !== prevProps.refocusWhenChange
+    if (didOpened) {
+      // Store a reference to the last active element, to restore it after
+      // modal close
+      this._lastFocusedElement = document.activeElement
+    }
+    if (shouldFocus) {
+      this.focusWrapper()
+    }
+
+    if (didClose) {
+      if (this._lastFocusedElement) {
+        this._lastFocusedElement.focus()
+      }
+    }
+  }
+
+  _wrapper = null
+  _lastFocusedElement = null
+
+  focusWrapper = () => {
+    // Forced to use findDOMNode here, because innerRef is giving a proxied component
+    const domWrapper = findDOMNode(this._wrapper) // eslint-disable-line react/no-find-dom-node
+
+    if (domWrapper instanceof HTMLDivElement) {
+      domWrapper.focus()
+    }
+  }
+
+  render() {
+    const { preventBackdropClick, isOpened, onHide, render, data, onClose, width } = this.props
+
+    return (
+      <Mortal
+        isOpened={isOpened}
+        onClose={onClose}
+        onHide={onHide}
+        closeOnEsc={!preventBackdropClick}
+        motionStyle={(spring, isVisible) => ({
+          opacity: spring(isVisible ? 1 : 0, springConfig),
+          scale: spring(isVisible ? 1 : 0.95, springConfig),
```

Comment:
is this approach efficient enough? Why not using CSS Transition?

Reply:
this is the old modal :p it will go away

Reply:
AH